### PR TITLE
upgrade backbone to 1.2.1

### PR DESCRIPTION
**Pull request by saponifi3d**
Update to all the niceness that is Backbone 1.2.1
... like the `update` event!!! :dancer:

### CODE REVIEW

File: test/client/base_view.test.js [hunk 1]
```diff
@@ -9,11 +9,18 @@ describe('Base/View', function () {
   var BaseView;
 
   before(function () {
+    global.document = {
+      createElement: sinon.stub().returns('<div></div>')
```
Comment (jmerrifield) [inline test/client/base_view.test.js]: Is this related to the backbone update?

Comment (saponifi3d) [inline test/client/base_view.test.js]: yep, the view constructor now requires document.createElement.  I figured this was the best way to make that happen. :)

**Comment by jmerrifield** [comment]
:+1:

**Comment by bigethan** [comment]
nice!

**Comment by saponifi3d** [comment]
@bigethan right?! i'm about to release 1.1.0 w/ that update and the emulateHTTP changes. this *is* a breaking change if you have an older version of underscore or lodash being required instead of underscore. (lodash 2.4.1 doesn't have _.matches, which is now in Backbone... found that out the hard way).

**Comment by pjanuario** [comment]
Nice!! But it sucks that lodash doesnt have that, since it's lighter than underscore.

**Comment by saponifi3d** [comment]
Pedro, latest version of lodash does have it, we are on v 2.4.0 which doesn't (latest is 3.9?) Sorry for the confusion!

**Comment by pjanuario** [comment]
@saponifi3d Thanks! Are you using lodash or underscore in production with rendr? Any issue with using lodash? It's on my plans to change that dependency. Is something special needed for that?

**Comment by saponifi3d** [comment]
@pjanuario no issues with lodash until now.. we have other issues that aren't allowing us to upgrade to latest of lodash. (Generally, haven't had time to go through and make sure that it will work correctly in 3.x since it's no longer backwards compatible).

We are using browserify to shim lodash everywhere that has `require('underscore')` to make sure we only deliver lodash to the client as well.. We've actually been doing this since day one of Rendr @ change.

**Comment by pjanuario** [comment]
:+1:
